### PR TITLE
Add video-call-reactions extension

### PR DESCRIPTION
**Pull request by yazanzaid00**
## Description

Upgrade your video calls on macOS with Raycast's "Video Call Reactions" extension. Choose from a suite of animated [reactions](https://support.apple.com/en-us/105117), including hearts and thumbs, to add a dynamic visual flair to your conversations. The confetti reaction stands out by activating Raycast's built-in confetti effect and an option to play a sound, enhancing your celebratory moments.

## Screencast

Check out all the reactions on [Imgur](https://imgur.com/a/ThwEeOd).

### 🎉 Confetti Celebration
gif-confetti

### 🎈 Joyful Balloons
balloons

### 🌧️ Soothing Rain
rain

### 👎 Disagree with Thumbs Down
thumbs-down

### 👍 Agree with Thumbs Up
thumbs-up

### 🎆 Festive Fireworks
fireworks

### ❤️ Lovely Hearts
hearts

### 🔆 Energetic Lasers
lasers

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by yazanzaid00** [comment]
I forgot to mention the last update also adds the option to play sound when confetti is triggered.

File: extensions/video-call-reactions/src/constants.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  }
+}
+
+// Preferences Interface
+export interface Preferences {
+  defaultReaction: string;
+  showConfetti: boolean;
+  playSound: boolean;
+}
+
```
Comment (pernielsentikaer) [inline extensions/video-call-reactions/src/constants.ts:65]:
```suggestion
```
This is already included in `raycast-env.d.ts`

File: extensions/video-call-reactions/package.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    "assets/balloons.png",
+    "assets/confetti.png",
+    "assets/hearts.png",
+    "assets/rain.png",
+    "assets/thumbs-up.png",
+    "assets/confetti-2.png",
+    "assets/fireworks.png",
+    "assets/lasers.png",
+    "assets/thumbs-down.png"
+  ]
```
Comment (pernielsentikaer) [inline extensions/video-call-reactions/package.json:119]: Whats the purpose of this?

File: extensions/video-call-reactions/package.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    "Other",
+    "Productivity",
+    "System",
+    "Web"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Trigger Video Reaction",
```
Comment (pernielsentikaer) [inline extensions/video-call-reactions/package.json:22]: I would maybe add an [argument](https://developers.raycast.com/information/lifecycle/arguments) to change it from the default one?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/video-call-reactions/package.json (hunk 2 above)
Comment (bigplayer-ai) [reply extensions/video-call-reactions/package.json:119]: I will remove those items. How can I add screenshots to the store?

Comment (yazanzaid00) [reply extensions/video-call-reactions/package.json:119]: I will remove those items. How can I add screenshots to the store?

File: extensions/video-call-reactions/src/constants.ts (hunk 1 above)
Comment (yazanzaid00) [reply extensions/video-call-reactions/src/constants.ts:65]: I used the getPreferenceValues from "@raycast/api" instead, thanks for mentioning that.

File: extensions/video-call-reactions/package.json (hunk 3 above)
Comment (yazanzaid00) [reply extensions/video-call-reactions/package.json:22]: I've included both the 'favorite reaction' and 'custom reaction' options, allowing users to swiftly toggle between two distinct choices. It's unclear to me whether the arguments retain their status upon closing or toggling, or if I need to implement this feature myself. 😊

Comment (pernielsentikaer) [reply extensions/video-call-reactions/package.json]: What about simply having just one command and use the arguments placeholder with the placeholder `Default Action` and then let the user switch to another?

Comment (yazanzaid00) [reply extensions/video-call-reactions/package.json]: > What about simply having just one command and use the arguments placeholder with the placeholder `Default Action` and then let the user switch to another?

Can the arguments save last state?

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**